frontend/lib/resolution/disambiguation.cpp

```diff
@@ -336,11 +335,11 @@ findMostSpecificCandidatesQuery(Context* context,
     int n = lst.size();
     for (int i = 0; i < n; i++) {
       QualifiedType forwardingTo;
-      if (!forwardingInfo.empty()) {
-        forwardingTo = forwardingInfo[i];
+      if (lst.hasForwardingInfo()) {
+        forwardingTo = lst.getForwardingInfo(i);
```

Comment:
invalid access if:

```C++
addCandidate(sig1);
addCandidate(sig2, &someType); // 2 signatures, 1 forwarding info -- forwardingInfoVec[1] is out of bounds.
```

Or if candidates were added and `helpComputeForwardingInfo` was not called. Assertions would probably suffice.

Reply:
If you allow types to be passed in `addCandidate`, you'll need to add some way of accounting for "gaps" in forwarding info. 

```
addCandidate(sig1, &someType);
addCandidate(sig2);
addCandidate(sig3, &someType);
```

Or you can remove the second argument to `addCandidate`, and there won't be gaps, but perhaps forwardingInfo will be shorter than candidates.

Reply:
Since you pointed out `addCandidate` is actually never used with `forwardingInfo`, I'm just going to remove it.